src/yam-sdk/utils/index.js

```diff
@@ -289,7 +306,11 @@ export const delegate = async (yam, account, onTxHash) => {
 };
 
 export const didDelegate = async (yam, account) => {
-  return (await yam.contracts.yamV3.methods.delegates(account).call()) === account;
+  const [unstaked, staked] = await Promise.allSettled([
+    yam.contracts.yamV3.methods.delegates(account).call(),
+    yam.contracts.voting_eth_pool.methods.delegates(account).call()
+  ]);
+  return unstaked === account || staked === account;
 };
```

Comment:
i think thats the issue, the account can (and would) be other than the current account, thats the idea behind delegating to another account